--- structure.sql	2017-01-20 17:33:49.000000000 +0900
+++ structure.rds.sql	2017-01-20 17:58:26.000000000 +0900
@@ -13,42 +13,25 @@
 SET client_min_messages = warning;
 SET row_security = off;
 
---
--- Name: postgis; Type: SCHEMA; Schema: -; Owner: -
---
-
-CREATE SCHEMA postgis;
-
-
---
--- Name: plpgsql; Type: EXTENSION; Schema: -; Owner: -
---
-
-CREATE EXTENSION IF NOT EXISTS plpgsql WITH SCHEMA pg_catalog;
-
-
---
--- Name: EXTENSION plpgsql; Type: COMMENT; Schema: -; Owner: -
---
-
-COMMENT ON EXTENSION plpgsql IS 'PL/pgSQL procedural language';
-
-
---
--- Name: postgis; Type: EXTENSION; Schema: -; Owner: -
---
-
-CREATE EXTENSION IF NOT EXISTS postgis WITH SCHEMA postgis;
-
-
---
--- Name: EXTENSION postgis; Type: COMMENT; Schema: -; Owner: -
---
-
-COMMENT ON EXTENSION postgis IS 'PostGIS geometry, geography, and raster spatial types and functions';
-
+--- From http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.PostgreSQL.CommonDBATasks.html#Appendix.PostgreSQL.CommonDBATasks.PostGIS
 
-SET search_path = public, pg_catalog;
+create extension postgis;
+create extension fuzzystrmatch;
+create extension postgis_tiger_geocoder;
+create extension postgis_topology;
+
+alter schema tiger owner to rds_superuser;
+alter schema topology owner to rds_superuser;
+
+CREATE FUNCTION exec(text) returns text language plpgsql volatile AS $f$ BEGIN EXECUTE $1; RETURN $1; END; $f$;
+
+SELECT exec('ALTER TABLE ' || quote_ident(s.nspname) || '.' || quote_ident(s.relname) || ' OWNER TO rds_superuser;')
+  FROM (
+    SELECT nspname, relname
+    FROM pg_class c JOIN pg_namespace n ON (c.relnamespace = n.oid)
+    WHERE nspname in ('tiger','topology') AND
+    relkind IN ('r','S','v') ORDER BY relkind = 'S')
+s;
 
 --
 -- Name: measurement_imports_subtype; Type: TYPE; Schema: public; Owner: -
@@ -130,7 +113,7 @@
     created_at timestamp without time zone DEFAULT '1970-01-01 00:00:00'::timestamp without time zone NOT NULL,
     updated_at timestamp without time zone DEFAULT '1970-01-01 00:00:00'::timestamp without time zone NOT NULL,
     bgeigie_import_id integer,
-    computed_location postgis.geography(Point,4326),
+    computed_location geography(Point,4326),
     md5sum character varying(255)
 );
 
@@ -284,7 +267,7 @@
     zoom_7_grid character varying(255),
     created_at timestamp without time zone NOT NULL,
     updated_at timestamp without time zone NOT NULL,
-    location postgis.geography(Point,4326),
+    location geography(Point,4326),
     md5sum character varying(255)
 );
 
@@ -455,7 +438,7 @@
     created_at timestamp without time zone NOT NULL,
     updated_at timestamp without time zone NOT NULL,
     unit character varying(255),
-    location postgis.geography(Point,4326),
+    location geography(Point,4326),
     location_name character varying(255),
     device_id integer,
     original_id integer,
